### PR TITLE
scheduler: clean up NoDiskConflict code in predicates.go

### PR DESCRIPTION
**Pull request by xiang90**
Just some clean up.

/cc @hongchaodeng @rootfs @HaiyangDING @davidopp

### CODE REVIEW

**Comment by rootfs** [comment]
@k8s-bot ok to test

**Comment by rootfs** [comment]
I like the code density but there is one scalability issue. In current format, the volume in question is checked against existing Pods only if this volume has either EBS, PD, or RBD. But the new code iterates through all Pods even the volume in question has none of such volume types. This could be a problem when there are a great many running Pods.

**Comment by xiang90** [comment]
@rootfs 

I noticed that when I writing the code. The overhead is that the targeted Vol has none of the three thing you want to check. Or you will have to do the iteration anyway. And the current code scales better if you have more than two types to check. I can add a short cut for the none case. But I wonder if it is necessary.

**Comment by rootfs** [comment]
The more than one types to check scenario is a good point but probably could be hard to find though :) , unless AWS and GCE could colocate.

**Comment by xiang90** [comment]
@rootfs I have no data to support any assumptions. Added a line of code to solve the issue you mentioned.

**Comment by rootfs** [comment]
new commit does the trick. thanks.

**Comment by xiang90** [comment]
@rootfs Thanks for the quick review and suggestion.

**Comment by HaiyangDING** [comment]
I think this is a good fix now.

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -73,46 +73,36 @@ func (c *CachedNodeInfo) GetNodeInfo(id string) (*api.Node, error) {
...
-	if volume.GCEPersistentDisk != nil {
-		disk := volume.GCEPersistentDisk
-
-		existingPod := &(pod.Spec)
-		for ix := range existingPod.Volumes {
-			if existingPod.Volumes[ix].GCEPersistentDisk != nil &&
-				existingPod.Volumes[ix].GCEPersistentDisk.PDName == disk.PDName &&
-				!(existingPod.Volumes[ix].GCEPersistentDisk.ReadOnly && disk.ReadOnly) {
+	// fast path if there is no conflict checking targets.
+	if volume.GCEPersistentDisk != nil && volume.AWSElasticBlockStore == nil && volume.RBD == nil {
```
Comment (davidopp) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: should this be
s/volume.GCEPersistentDisk != nil/volume.GCEPersistentDisk == nil/
?

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 2]
```diff
@@ -73,46 +73,36 @@ func (c *CachedNodeInfo) GetNodeInfo(id string) (*api.Node, error) {
...
+		return false
+	}
+
+	for _, existingVolume := range pod.Spec.Volumes {
+		// Same GCE Disk can be mounted as read-only by multiple pod simultaneously.
```
Comment (davidopp) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Slightly clearer comment: "Same GCE disk mounted by multiple pods conflicts unless all pods mount it read-only."

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 1 above)
Comment (hongchaodeng) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: @xiang90

**Comment by xiang90** [comment]
@hongchaodeng @davidopp Sorry for the delay. All fixed. PTAL.

**Comment by davidopp** [comment]
LGTM

**Comment by alex-mohr** [comment]
@k8s-bot unit test this

**Pull request merged**